extract.go

```diff
@@ -45,36 +46,36 @@ func extractOne(ti TarInterpreter, s io.Reader) error {
 
 // Ensures that file extension is valid. Any subsequent behavior
 // depends on file type.
-func tarHandler(writeCloser io.WriteCloser, readerMaker ReaderMaker, crypter Crypter) error {
+func handleTar(writeCloser io.WriteCloser, readerMaker ReaderMaker, crypter Crypter) error {
 	defer writeCloser.Close()
-	r, err := readerMaker.Reader()
+	readCloser, err := readerMaker.Reader()
 
 	if err != nil {
 		return errors.Wrap(err, "ExtractAll: failed to create new reader")
 	}
-	defer r.Close()
+	defer readCloser.Close()
 
 	if crypter.IsUsed() {
 		var reader io.Reader
-		reader, err = crypter.Decrypt(r)
+		reader, err = crypter.Decrypt(readCloser)
 		if err != nil {
 			return errors.Wrap(err, "ExtractAll: decrypt failed")
 		}
-		r = ReadCascadeClose{reader, r}
+		readCloser = ReadCascadeCloser{reader, readCloser}
 	}
 
-	if readerMaker.Format() == "lzo" {
-		err = DecompressLzo(writeCloser, r)
-		if err != nil {
-			return errors.Wrap(err, "ExtractAll: lzo decompress failed. Is archive encrypted?")
+	for _, decompressor := range Decompressors {
+		if readerMaker.Format() != decompressor.FileExtension() {
+			continue
 		}
-	} else if readerMaker.Format() == Lz4FileExtension {
-		_, err = DecompressLz4(writeCloser, r)
+		err = decompressor.Decompress(writeCloser, readCloser)
 		if err != nil {
-			return errors.Wrap(err, "ExtractAll: lz4 decompress failed. Is archive encrypted?")
+			return errors.Wrap(err, fmt.Sprintf("ExtractAll: %v decompress failed. Is archive encrypted?", decompressor.FileExtension()))
```

Comment:
How about errors.Wrapf? If it's our package

Reply:
Fixed